crates/ursa-rpc-service/src/api.rs

```diff
@@ -283,6 +241,177 @@ where
     }
 }
 
+impl<S> NodeNetworkInterface<S>
+where
+    S: Blockstore + Store + Send + Sync + 'static,
+{
+    pub fn new(
+        store: Arc<UrsaStore<S>>,
+        network_send: Sender<NetworkCommand>,
+        provider_send: Sender<ProviderCommand>,
+        origin_config: OriginConfig,
+    ) -> Self {
+        Self {
+            store,
+            network_send,
+            provider_send,
+            origin_config,
+            pending_requests: Arc::new(RwLock::new(HashMap::new())),
+            client: Arc::new(Client::new()),
+        }
+    }
+
+    /// Ensure a root cid is synced to the blockstore
+    async fn sync_content(&self, cid: Cid) -> Result<()> {
+        if !self.store.blockstore().has(&cid)? {
+            info!("Requesting block with the cid {cid:?}");
+            if let Err(e) = self.get_network(cid).await {
+                info!("Failed to get content from network: {}", e);
+                let size = self.get_origin(cid).await?;
+                self.provide_cid(cid, Some(size)).await
+            } else {
+                self.provide_cid(cid, None).await
+            }
+        } else {
+            Ok(())
+        }
```

Comment:
```suggestion
        if !self.store.blockstore().has(&cid)? {
            info!("Requesting block with the cid {cid:?}");
            if let Err(e) = self.get_network(cid).await {
                info!("Failed to get content from network: {}", e);
                let size = self.get_origin(cid).await?;
            } 
            self.provide_cid(cid, Some(size)).await
        }
        Ok(())
```

if the case above has to hold something like this where we have to enforce size

Reply:
size wouldn't be defined if bitswap is successful

Reply:
I am not sure i follow, size is always defined no?

Reply:
Bitswap doesn't return any information about the car size of the content it fetched

In the code you suggested, size only exists in the scope of if bitswap failed and get origin succeeded

Reply:
Still need more information why would bitswap return car file size?

Reply:
In the code I suggest that size should always be some not none

Reply:
yup, realized what you were suggesting, will make the change to the method and move the logic

Reply:
https://github.com/fleek-network/ursa/pull/278/commits/171e54140367661a9e63e432d178b3ed9b0f876a